File: bench/go/fileio.go
package main

import (
	"fmt"
	"os"
	"time"
)

func main() {
	if len(os.Args) < 2 {
		fmt.Fprintln(os.Stderr, "Usage: fileio <filepath>")
		os.Exit(1)
	}

	// Read file outside timer
	buf, err := os.ReadFile(os.Args[1])
	if err != nil {
		fmt.Fprintf(os.Stderr, "Cannot open %s: %v\n", os.Args[1], err)
		os.Exit(1)
	}

	start := time.Now()

	words := 0
	inWord := false
	for _, b := range buf {
		if b == ' ' || b == '\t' || b == '\n' || b == '\r' || b == '\x0b' || b == '\x0c' {
			inWord = false
		} else {
			if !inWord {
				words++
				inWord = true
			}
		}
	}

	elapsed := time.Since(start).Seconds() * 1000

	fmt.Printf("words: %d\n", words)
	fmt.Printf("RESULT:%.3f\n", elapsed)
}

File: bench/python/fileio.py
import sys
import time

if len(sys.argv) < 2:
    print("Usage: python fileio.py <filepath>", file=sys.stderr)
    sys.exit(1)

# Read file outside timer
with open(sys.argv[1], "rb") as f:
    buf = f.read()

start = time.perf_counter()

words = 0
in_word = False
for b in buf:
    if b in (0x20, 0x09, 0x0A, 0x0D, 0x0B, 0x0C):  # isspace
        in_word = False
    else:
        if not in_word:
            words += 1
            in_word = True

end = time.perf_counter()
ms = (end - start) * 1000

print(f"words: {words}")
print(f"RESULT:{ms:.3f}")
